### ISSUE TITLE
trace_block(46169,46170) failed of insufficient funds for gas * price + value

### ISSUE THREAD

**Issue by jsvisa** [paradigmxyz/reth#3427]
### Describe the bug

My reth node is fully synced, and I'm testing it with `trace_block` requests, seems some blocks are failed, currently failed for those blocks:

- https://etherscan.io/block/46169
- https://etherscan.io/block/46170

AFAIK, the same error also appeared in [erigon](https://github.com/ledgerwatch/erigon/issues?q=is%3Aissue+is%3Aopen+insufficient+funds+for+gas+*+price+%2B+value+) multi times, and not fully fixed by now.

### Steps to reproduce

> request

```
curl -XPOST 127.0.0.1:5545 -H 'content-type: application/json' -d '{"jsonrpc":"2.0","id":1,"method":"trace_block","params":["0xB459"]}' -v
```

> response

```json
{"jsonrpc":"2.0","error":{"code":-32003,"message":"insufficient funds for gas * price + value"},"id":1}
```

### Node logs

```text
None
```

### Platform(s)

Linux (x86)

### What version/commit are you on?

d3d44fd46f03481ae0ab42adc5e6334ee64cd3b5

### If you've built Reth from source, provide the full command you used

docker build 

### Code of Conduct

- [X] I agree to follow the Code of Conduct

**Comment by mattsse** [comment]
thanks for these, I'll debug them could be related to old hardfork rules perhaps

cc @rakita

**Comment by jsvisa** [comment]
@mattsse btw, the Parity's trace_block will return block reward records, is this feature also implemented in reth?

**Comment by mattsse** [comment]
this wasn't in the original spec, does erigon return them?

do you have an example for this? please open a separate issue for this

**Comment by mattsse** [comment]
I think it actually is:

single tx in block, sender is genesis acc:

https://etherscan.io/address/0xbd08e0cddec097db7901ea819a3d1fd9de8951a2

**Comment by jsvisa** [comment]
> I think it actually is:
> 
> bot blocks have single tx in block, sender is genesis acc:
> 
> https://etherscan.io/address/0xbd08e0cddec097db7901ea819a3d1fd9de8951a2
> 
> closing this because was fixed in #3346

Thanks, I'll have another test

**Comment by jsvisa** [comment]
> I think it actually is:
> 
> bot blocks have single tx in block, sender is genesis acc:
> 
> https://etherscan.io/address/0xbd08e0cddec097db7901ea819a3d1fd9de8951a2
> 
> closing this because was fixed in #3346

@mattsse  @rakita I used the latest commit bb57556a6ab84f1c9f09eebd1f618220430b5101 to re-run reth, but I still have this problem. I guess it may need to resync to store them on the disk, but in this way it is too much to lose. Are there any db tools that can quickly repair the similar database errors?
